### DISCUSSION TITLE
Android NDK is not  installed

### DISCUSSION THREAD

**Discussion by apolloandroid** [strongswan/strongswan#778] [Q&A]
Hello!
I am trying to develop android app using your library.
I configured files from your repository. And now trying to build app with this files in Android studio. But every time Build fails with error message: NDK not installed. However, I have definitely installed NDK and set path in gradle local.properties.
OS: Ubuntu.
Gradle plugin: 4.1.1.

**Comment by tobiasbrunner** [comment]
Sorry, can't help you. You might better ask this somewhere people actually know anything about Android Studio (I'm a mere user myself, and a very infrequent one at that).

**Comment by realbitt** [comment]
first of all u can't compile strongswan source in windows, so u need Linux destro. I've used ubunto 20.04 witj GUI .
then
**1- download strongswan source** 
**2- install these packages in your linux**
gcc
automake
autoconf
libtool
pkg-config
gettext
perl
python
flex
bison
gperf
libgmp-dev
**3- access strongswan folder** 
4-  Run > ./autogen.sh && ./configure && make dist
5- access this folder > src/frontends/android/app/src/main/jni
6- download ssl lib > git clone git://git.strongswan.org/android-ndk-boringssl.git -b ndk-static openssl
7- install android studio & cmake & ndk  version 20.xx
8- modify build.gradle in app folder and add > ndkVersion "20.xx" inside android section
8- build app

**Reply by apolloandroid** [reply]
I have already done this steps. But it did not help(

**Reply by realbitt** [reply]
Make sure ndk path defined in system configuration file.

**Reply by apolloandroid** [reply]
already

**Reply by realbitt** [reply]
copy error log here

**Reply by apolloandroid** [reply]
![photo_2021-11-22_11-55-59 (2)](https://user-images.githubusercontent.com/62345510/143308189-f233459d-d61c-47cb-8ff6-abf5cbf1c915.jpg)

**Reply by realbitt** [reply]
may I can help you over anydesk.

**Reply by apolloandroid** [reply]
Yes, but not now. What time zone are you in?

**Reply by realbitt** [reply]
its 11:25 PM Midnight , GMT+2

**Reply by apolloandroid** [reply]
How about tomorrow at 19:00 your time?

**Reply by realbitt** [reply]
ok

**Reply by apolloandroid** [reply]
please write your email, i will send my anydesk address

**Reply by apolloandroid** [reply]
done

**Reply by MahdiAhmadnejad** [reply]
what was the problem?
what have you done?
